tests/Test.ADAL.NET.Unit.net45/TokenCacheUnitTests.cs

```diff
@@ -186,11 +188,72 @@ public void ParallelStoreTest()
         }
 
         [TestMethod]
-        [Description("Test to ensure the token cache doesnt throw an exception when cleared")]
+        [Description("Test to ensure the token cache doesn't throw an exception when cleared")]
         [TestCategory("AdalDotNet.Unit")]
         public void TokenCacheClearTest()
         {
             TokenCacheTests.TokenCacheClearTest(File.ReadAllBytes("oldcache.serialized"));
         }
+
+        [TestMethod]
+        [Description("Test to ensure a null IdToken is handled correctly")]
+        [TestCategory("AdalDotNet.Unit")]
+        public void NullIdTokenCacheTest()
+        {
+            // arrange & act
+            TokenCache tokenCache = new TokenCache();
+
+            WriteMsalRefreshToken(CreateAdalResultWrapper(), tokenCache.TokenCacheAccessor);
+
+            // assert
+            // no IdToken present, CacheFallbackOperations exits early
+            Assert.AreEqual(tokenCache.TokenCacheAccessor.AccountCount, 0);
+            Assert.AreEqual(tokenCache.TokenCacheAccessor.RefreshTokenCount, 0);
+        }
+
+        [TestMethod]
+        [Description("Test to ensure WriteMsalRefreshToken handles presence of IdToken correctly")]
+        [TestCategory("AdalDotNet.Unit")]
+        public void IdTokenReturnedCacheTest()
+        {
+            // arrange
+            TokenCache tokenCache = new TokenCache();
+
+            var result = CreateAdalResultWrapper();
+
+            result.Result.IdToken = "some-id-token";
```

Comment:
>result.Result.IdToken = "some-id-token"; [](start = 12, length = 40)

perhaps something to be passed into the wrapper? Or another method which does this? seems like magic what you are doing here, this can be better explained with a method names elegantly imo